go/integrityblock/hash.go

```diff
@@ -0,0 +1,52 @@
+package integrityblock
+
+import (
+	"crypto/sha512"
+	"errors"
+	"io"
+	"os"
+)
+
+const hash512ChunkSize int64 = 8192
+
+var hash512ChunkSizeForTesting int64 = 0
+
+func GetHash512ChunkSize() int64 {
+	if hash512ChunkSizeForTesting > 0 {
+		return hash512ChunkSizeForTesting
+	}
+	return hash512ChunkSize
+}
+func SetHash512ChunkSizeForTesting(val int64) {
+	hash512ChunkSizeForTesting = val
+}
+
+// ComputeWebBundleSha512 computes the SHA-512 hash over the given web bundle file with the given chunk size. chunkSize is for improved testability.
+func ComputeWebBundleSha512(bundleFile *os.File, offset int64) ([]byte, error) {
```

Comment:
I think this should work:
```go
func ComputeWebBundleSha512(bundleFile *os.File, offset int64) ([]byte, error) {
	bundleFile.Seek(offset, io.SeekStart)
	h := sha512.New()
	_, err := io.Copy(h, bundleFile)
	if err != nil {
		return nil, err
	}
	return h.Sum(nil), nil
}
```
The `io.Copy()` will do chunked read/write under the hood.